### ISSUE TITLE
Terminal.resizeLazy function

### ISSUE THREAD

**Issue by jerch** [xtermjs/xterm.js#4114]
Idea to relief some stress from terminal reflow events - create a `Terminal.resizeLazy(rows, cols, timeout?)` variant, and likewise a `FitAddon.fitLazy` variant. This would help during window resizing to avoid processing every single number change, and also lower the back-messaging and reflow stress on application side (less SIGWINCH signals).

**Comment by Tyriar** [comment]
I think the resize experience is best done by the embedder as it's easy to wrap `resize` calls in a debouncer. We could recommend this in the .d.ts docs though?

**Comment by jerch** [comment]
> I think the resize experience is best done by the embedder as it's easy to wrap `resize` calls in a debouncer. We could recommend this in the .d.ts docs though?

Yes good idea. While I think that this is best left to embedders, most integrations I've came across dont care at all. So not sure, if a doc note will kick enough to get devs out of their comfort zone, lol.

**Comment by Tyriar** [comment]
How about we put a simple debouncer into client.ts as well as the docs?

It's fine if people don't care so much about their projects 🤷 it may not be important for them.

**Comment by jerch** [comment]
Changed the title - so lets do a resize debouncer in client.ts + docs about it.
